app/models/post.rb

```diff
@@ -1,18 +1,22 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
-  has_many :comments
-  has_many :likes
-  after_save :post_counter
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :likes, foreign_key: :post_id, dependent: :destroy
+  has_many :comments, foreign_key: :post_id, dependent: :destroy
 
   validates :title, presence: true, length: { minimum: 3, maximum: 250 }
-  validates :likes_counters, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :likes_counters, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  def post_counter
-    author.increment!(:posts_counter)
-  end
+  after_save :post_counter
 
   def recent_comments
-    comments.order(created_at: :desc).limit(5)
+    # comments.order(created_at: :desc).limit(5)
```

Comment:
- [OPTIONAL] Same here, please remove all unnecessary titles since they are empty  or add details to them (Prerequisites, run tests Deployment ....) 🚮